src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -198,6 +198,9 @@ private final void closeStatements()
             }
             catch (SQLException e) {
                checkException(e);
+               if (delegate == ClosedConnection.CLOSED_CONNECTION) {
+                  break; //connection closed in checkException
```

Comment:
I know this is more efficient, but it is optimizing a case so rare that I really don't care about the performance.

The additional code just adds cognitive load to an already messy method.